kratos/modified_shape_functions/modified_shape_functions.cpp

```diff
@@ -0,0 +1,359 @@
+        std::vector<Matrix> &rPositiveSideShapeFunctionsGradientsValues,
+        Vector &rPositiveSideWeightsValues,
+        const IntegrationMethodType IntegrationMethod) {
+
+        KRATOS_ERROR << "Calling the base class GetShapeFunctionsAndGradientsValuesPositiveSide method. Call the specific geometry one.";
+    };
+
+    // Computes all the shape function values for the negative side.
+    void ModifiedShapeFunctions::GetNegativeSideShapeFunctionsAndGradientsValues(
+        Matrix &rNegativeSideShapeFunctionsValues,
+        std::vector<Matrix> &rNegativeSideShapeFunctionsGradientsValues,
+        Vector &rNegativeSideWeightsValues,
+        const IntegrationMethodType IntegrationMethod) {
+
+        KRATOS_ERROR << "Calling the base class GetShapeFunctionsAndGradientsValuesNegativeSide method. Call the specific geometry one.";
+    };
+
+    // Computes all the shape function values for the positive interface side.
+    void ModifiedShapeFunctions::GetInterfacePositiveSideShapeFunctionsAndGradientsValues(
+        Matrix &rInterfacePositiveSideShapeFunctionsValues,
+        std::vector<Matrix> &rPositiveInterfaceSideShapeFunctionsGradientsValues,
+        Vector &rPositiveInterfaceSideWeightsValues,
+        const IntegrationMethodType IntegrationMethod) {
+
+        KRATOS_ERROR << "Calling the base class GetInterfaceShapeFunctionsAndGradientsValuesPositiveSide method. Call the specific geometry one.";
+    };
+
+    // Computes all the shape function values for the negative interface side.
+    void ModifiedShapeFunctions::GetInterfaceNegativeSideShapeFunctionsAndGradientsValues(
+        Matrix &rInterfaceNegativeSideShapeFunctionsValues,
+        std::vector<Matrix> &rInterfaceNegativeSideShapeFunctionsGradientsValues,
+        Vector &rInterfaceNegativeSideWeightsValues,
+        const IntegrationMethodType IntegrationMethod) {
+
+        KRATOS_ERROR << "Calling the base class GetInterfaceShapeFunctionsAndGradientsValuesNegativeSide method. Call the specific geometry one.";
+    };
+
+    // Computes the positive side interface outwards unit normal vector values.
+    void ModifiedShapeFunctions::GetPositiveSideInterfaceUnitNormals(
+        std::vector<Vector> &rPositiveSideInterfaceUnitNormal,
+        const IntegrationMethodType IntegrationMethod) {
+
+        KRATOS_ERROR << "Calling the base class GetPositiveSideInterfaceUnitNormals method. Call the specific geometry one.";
+    };
+
+    // Computes the positive side interface outwards unit normal vector values.
+    void ModifiedShapeFunctions::GetNegativeSideInterfaceUnitNormals(
+        std::vector<Vector> &rNegativeSideInterfaceUnitNormal,
+        const IntegrationMethodType IntegrationMethod) {
+
+        KRATOS_ERROR << "Calling the base class GetNegativeSideInterfaceUnitNormals method. Call the specific geometry one.";
+    };
+
+    // Returns true if the element is splitting
+    bool ModifiedShapeFunctions::IsSplit() {
```

Comment:
do you need to make this virtual? i guess that the check is identical for all the derived classes no?

Reply:
The check is the same (return the bool member variable mIsSplit). However, this variable might be taken from different geometry splitting utility depending on the geometry type. Thus, the method is different for each derived class.